server/plugin.go

```diff
@@ -0,0 +1,373 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/plugin"
+)
+
+type Plugin struct {
+	plugin.MattermostPlugin
+
+	// configurationLock synchronizes access to the configuration.
+	configurationLock sync.RWMutex
+
+	// configuration is the active plugin configuration. Consult getConfiguration and
+	// setConfiguration for usage.
+	configuration *configuration
+
+	TeamID    string
+	ChannelID string
+	BotUserID string
+}
+
+const (
+	SNS_ICON_URL = "https://cdn2.iconfinder.com/data/icons/amazon-aws-stencils/100/App_Services_copy_Amazon_SNS-512.png"
+	SNS_USERNAME = "AWS SNS Bot"
+)
+
+func (p *Plugin) OnActivate() error {
+	configuration := p.getConfiguration()
+
+	if err := p.IsValid(configuration); err != nil {
+		return err
+	}
+
+	split := strings.Split(p.configuration.TeamChannel, ",")
+	teamSplit := split[0]
+	channelSplit := split[1]
+
+	team, err := p.API.GetTeamByName(teamSplit)
+	if err != nil {
+		return err
+	}
+	p.TeamID = team.Id
+
+	user, err := p.API.GetUserByUsername(p.configuration.Username)
+	if err != nil {
+		p.API.LogError(err.Error())
+		return fmt.Errorf("Unable to find user with configured username: %v", p.configuration.Username)
+	}
+	p.BotUserID = user.Id
+
+	channel, err := p.API.GetChannelByName(team.Id, channelSplit, false)
+	if err != nil && err.StatusCode == http.StatusNotFound {
+		channelToCreate := &model.Channel{
+			Name:        channelSplit,
+			DisplayName: channelSplit,
+			Type:        model.CHANNEL_OPEN,
+			TeamId:      p.TeamID,
+			CreatorId:   p.BotUserID,
+		}
+
+		newChannel, errChannel := p.API.CreateChannel(channelToCreate)
+		if err != nil {
```

Comment:
There's a problem here. I think he meant `errChannel != nil`. 
`err` will always be `!= nil` here.